mlflow/models/utils.py

```diff
@@ -97,18 +94,31 @@ def _is_scalar(x):
                 "(pandas.DataFrame, numpy.ndarray, dict, list), "
                 "got {}".format(type(input_example))
             )
+
         example_filename = "input_example.json"
-        self.data = input_example.to_dict(orient="split")
-        # Do not include row index
-        del self.data["index"]
-        if all(input_example.columns == range(len(input_example.columns))):
-            # No need to write default column index out
-            del self.data["columns"]
-        self.info = {
-            "artifact_path": example_filename,
-            "type": "dataframe",
-            "pandas_orient": "split",
-        }
+        if isinstance(input_example, (np.ndarray, dict)):
+            if isinstance(input_example, np.ndarray):
+                input_example = input_example.tolist()
+                self.data = {"instances": input_example}
```

Comment:
should we use inputs in for both numpy arrays and dictionaries for consistency? I don't think it makes a difference in case of a single tensor, but it may be easier for users to read the json if we always stick to one format.

Reply:
This might get a little weird since the `inputs` format always expects a dictionary. https://github.com/mlflow/mlflow/blob/master/mlflow/pyfunc/scoring_server/__init__.py#L200
Thus for a single tensor, we will have to store it as a dict. Again while reading the json, read the dict and return the single example (otherwise it wont work as-is).

Plus, we have the utility to load the tf-serving format which can be used.